src/SourceTerms/LightBulb/CPU_Src_Lightbulb.cpp

```diff
@@ -198,14 +209,26 @@ static void Src_Lightbulb( real fluid[], const real B[],
    const real rate_Code = ( rate_heating - rate_cooling ) * ( Xn + Xp ) * EXP( -tau );
 
 
-// 3. calculate the change in internal energy and update the input energy density
+// 3. calculate the change in internal energy and update the input energy density and temperature for initial guess
    const real dEint_Code  = rate_Code * dt * Dens_Code;
    const real Eint_Update = Eint_Code + dEint_Code;
 
    fluid[ENGY] = Hydro_ConEint2Etot( fluid[DENS], fluid[MOMX], fluid[MOMY], fluid[MOMZ], Eint_Update, Emag );
 #  ifdef DEDT_LB
    fluid[DEDT_LB] = FABS( rate_Code * Dens_Code );
 #  endif
+// calculate temperature for initial guess and store to fluid[TEMP_IG]
+#  ifdef TEMP_IG
+#  ifdef __CUDACC__
+   fluid[TEMP_IG] = Hydro_Con2Temp( fluid[DENS], fluid[MOMX], fluid[MOMY], fluid[MOMZ], fluid[ENGY], fluid+NCOMP_FLUID,
+                                    false, 0.0, Emag, EoS->DensEint2Temp_FuncPtr,
+                                    EoS->AuxArrayDevPtr_Flt, EoS->AuxArrayDevPtr_Int, EoS->Table );
+#  else
+   fluid[TEMP_IG] = Hydro_Con2Temp( fluid[DENS], fluid[MOMX], fluid[MOMY], fluid[MOMZ], fluid[ENGY], fluid+NCOMP_FLUID,
+                                    false, 0.0, Emag, EoS_DensEint2Temp_CPUPtr,
+                                    EoS_AuxArray_Flt, EoS_AuxArray_Int, h_EoS_Table );
+#  endif
+#  endif
```

Comment:
Why do we need to invoke `Hydro_Con2Temp()` which re-compute the internal energy and Ye? Why not update `In_Flt[1] = Eint_Update`and call the `EoS->General_CPU/GPUPtr` directly?